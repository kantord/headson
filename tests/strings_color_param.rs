use insta::assert_snapshot;

fn run_color(input: &str, template: &str) -> String {
    let mut cmd = assert_cmd::cargo::cargo_bin_cmd!("headson");
    let mut args = vec!["--color", "-n", "1000"];
    let lower = template.to_ascii_lowercase();
    match lower.as_str() {
        "json" => args.extend(["-f", "json", "-t", "strict"]),
        "pseudo" => args.extend(["-f", "json", "-t", "default"]),
        "js" => args.extend(["-f", "json", "-t", "detailed"]),
        other => args.extend(["-f", other]),
    }
    let assert = cmd.args(args).write_stdin(input).assert().success();
    String::from_utf8_lossy(&assert.get_output().stdout).into_owned()
}

#[test]
fn color_string_across_templates() {
    let input = "\"hello\"";
    for tmpl in ["json", "pseudo", "js"] {
        let out = run_color(input, tmpl);
        assert_snapshot!(format!("color_string_{}", tmpl), out);
    }
}

#[test]
fn color_object_key_and_value_across_templates() {
    let input = "{\"k\":\"v\"}";
    for tmpl in ["json", "pseudo", "js"] {
        let out = run_color(input, tmpl);
        assert_snapshot!(format!("color_object_kv_{}", tmpl), out);
    }
}
