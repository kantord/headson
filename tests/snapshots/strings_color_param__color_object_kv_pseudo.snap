---
source: tests/strings_color_param.rs
expression: "color_object_kv_pseudo"
---
{
  [34m"k"[0m: [34m"v"[0m
}

